app/api/events.py

```diff
@@ -115,78 +126,214 @@ def update_event(request: Request, id:str, eventUpdate: EventUpdate, AccessToken
+    participant = participant.dict()
+
+    db.events.update_one({'eid': event['eid']}, {
+        "$addToSet": {"participants": participant}})
+
+    # Add to the participants list regardess. SubmitDate functions as prioritization filter.
 
     if event['maxParticipants']:
         if len(event['participants']) >= event['maxParticipants']:
             return {'max': True}
 
     return {'max': False}
 
+
 @router.post('/{id}/leave', dependencies=[Depends(validate_uuid)])
-def leave_event(request: Request, id:str, token: AccessTokenPayload = Depends(authorize)):
+def leave_event(request: Request, id: str, token: AccessTokenPayload = Depends(authorize)):
     db = get_database(request)
-
     event = get_event_or_404(db, id)
-    
     member = db.members.find_one({'id': token.user_id})
 
     if not member:
         raise HTTPException(400, "User could not be found")
 
-    user = db.events.find_one({"eid" : event["eid"]}, {"participants" : {"$elemMatch" : {"id": token.user_id}}})
-    if not user:
+    participant = db.events.find_one({"eid": event["eid"]}, {"participants": {
+        "$elemMatch": {"id": token.user_id}}})
+
+    if not participant:
         raise HTTPException(400, "User not joined event!")
 
-    try : 
-        user['participants'][0]
+    try:
+        participant['participants'][0]
     except KeyError:
         raise HTTPException(400, "User not joined event!")
 
-    db.events.update_one({'eid' : event['eid']}, { "$pull": { "participants": member } })
+    db.events.update_one({'eid': event['eid']}, {
+                         "$pull": {"participants": {"id": token.user_id}}})
+
+    return {'max': False}
 
-    return {'max': event['maxParticipants'] }
 
 @router.get('/{id}/joined', dependencies=[Depends(validate_uuid)])
-def is_joined_event(request: Request, id:str, token: AccessTokenPayload = Depends(authorize)):
+def is_joined_event(request: Request, id: str, token: AccessTokenPayload = Depends(authorize)):
     db = get_database(request)
 
     event = get_event_or_404(db, id)
-    # uses event["eid"] instead of casting eid -> UUID 
-    user = db.events.find_one({"eid" : event["eid"]}, {"participants" : {"$elemMatch" : {"id": token.user_id}}})
+    # uses event["eid"] instead of casting eid -> UUID
+    user = db.events.find_one({"eid": event["eid"]}, {"participants": {
+                              "$elemMatch": {"id": token.user_id}}})
 
     if not user:
-        return {'joined' : False}
+        return {'joined': False}
 
-    try : 
+    try:
         user['participants'][0]
-        return {'joined' : True}
+        return {'joined': True}
     except KeyError:
-        return {'joined' : False}
+        return {'joined': False}
+
+
+@router.delete('/{id}/removeParticipant/{uid}', dependencies=[Depends(validate_uuid)])
+def leave_event(request: Request, id: str, uid: str, token: AccessTokenPayload = Depends(authorize)):
+    role_required(token, "admin")
```

Comment:
if the endpoint requires admin, it should be Depends(authorize_admin) as role_required is no longer needed